app/models/exercise.rb

```diff
@@ -6,4 +6,8 @@ class Exercise < ApplicationRecord
   has_many :solutions
   has_many :exercise_tags
   has_many :tags, through: :exercise_tags
+
+  def user_solution?(user)
+    solutions.where(user: user).count > 0
```

Comment:
This took me a minute to figure out but I dig it. Maybe in the future we can rename it something like `does_exercise_have_solution?`